modules/auxiliary/admin/mssql/mssql_findandsampledata.rb

```diff
@@ -342,7 +344,13 @@ def sql_statement()
 
     # CREATE DATABASE CONNECTION AND SUBMIT QUERY WITH ERROR HANDLING
     begin
-      result = mssql_query(sql, false) if mssql_login_datastore
+      if session
+        set_session(session.client)
+        result = mssql_query(sql, false)
+      elsif mssql_login_datastore
+        result = mssql_query(sql, false)
+      end
```

Comment:
Could we just update this to:

```suggestion
      set_session(session.client)  if session
      
      result = mssql_query(sql, false)
```

Am I right in assuming the outcome of the if statement is not actually affecting whats being done with `mssql_query`, doesn't look like something is conditionally being passed as a argument based on if we have a session or not?

Reply:
Also, I believe we'd want to return here if we don't have either a `session` or `mssql_login_datastore` fails to log in.

```suggestion
      if session
        set_session(session.client)
      else 
        return unless mssql_login_datastore
      end
```

Reply:
Potentially worth double checking this in other modules as well.